wallet/wallet_interface.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package wallet
+
+import (
+	"context"
+
+	"github.com/ava-labs/avalanche-tooling-sdk-go/account"
+	"github.com/ava-labs/avalanche-tooling-sdk-go/tx"
+
+	"github.com/ava-labs/avalanche-tooling-sdk-go/network"
+	"github.com/ava-labs/avalanchego/ids"
+)
+
+type BuildTxParams struct {
+	BuildTxInput
+	Account account.Account
+	Network network.Network
+}
+
+// BuildTxInput represents a generic interface for transaction parameters
+type BuildTxInput interface {
+	// GetTxType returns the transaction type identifier
+	GetTxType() string
+	// Validate validates the parameters
+	Validate() error
+	// GetChainType returns which chain this transaction is for
+	GetChainType() string
+}
+
+type SignTxParams struct {
+	*tx.BuildTxResult
+	Account account.Account
+	Network network.Network
+}
+
+type SendTxParams struct {
+	*tx.SignTxResult
+	Account account.Account
+	Network network.Network
+}
+
+// Wallet represents the core wallet interface that can be implemented
+// by different wallet types (local, API-based, etc.)
+type Wallet interface {
+	// Accounts returns the accounts in the Wallet
+	Accounts() []account.Account
+
+	// Signer returns the clients in the Wallet
+	Clients() ChainClients
+	// Account Management
+	// CreateAccount creates a new Account
+	CreateAccount(ctx context.Context) (*account.Account, error)
+
+	// GetAccount retrieves an existing Account by address or identifier
```

Comment:
What does identifier mean here?

Reply:
identifier can be a human readable name given to an account, similar to https://docs.cdp.coinbase.com/server-wallets/v2/using-the-wallet-api/managing-accounts

Reply:
Input args takes only a type ids.ShortID. Should this be extended to allow a string identifier?